src/main/java/de/doubleslash/keeptime/view/SettingsController.java

```diff
@@ -239,6 +243,56 @@ private void initialize() {
          aboutStage.show();
       });
    }
+   private void initImportButton(){
+      LOG.debug("Initialize importButton.");
+      importButton.setOnAction(event ->{
+
+         try {
+            Alert confirmationAlert = new Alert(AlertType.CONFIRMATION , "", ButtonType.YES, ButtonType.NO);
+            confirmationAlert.setTitle("Import");
+            confirmationAlert.setHeaderText("Do you want to Override current Data ?");
+            confirmationAlert.showAndWait();
+
+            if(confirmationAlert.getResult()==ButtonType.NO){
+               LOG.info("User canceled import");
+               return;
+            }
+
+            final FileChooser fileChooser = new FileChooser();
+            fileChooser.setTitle("Open Exported SQl Script");
+            File file = fileChooser.showOpenDialog(thisStage);
+            if (file == null) {
+               LOG.info("User canceled import.");
+               return;
+            }
+            LOG.info(file.toString());
+
+            final String url = applicationProperties.getSpringDataSourceUrl();
+            final String username = applicationProperties.getSpringDataSourceUserName();
+            final String password = applicationProperties.getSpringDataSourcePassword();
+            RunScript.execute(url, username, password, file.toString(), Charset.defaultCharset(), true);

```

Comment:
[H2 Migration to v2](https://www.h2database.com/html/migration-to-v2.html) states that we may need to specify `FROM_1X flag`. Do we need that?

Reply:
As far as I tested it it worked without the tag.

Reply:
the docs state "Use this clause when you import script that was generated by H2 1.4.200 or an older version into more recent version.". As this is the case for us we should set the flag.

Later (if another upgrade is needed) we may need to validate the version we want to import (to set the flag if we come from 1.x and dont set it from 2.x).

Reply:
Added the option

Reply:
can you add a // TODO above the RunScript, to add an option at the next release to make the "FROM_1X flag" configureable. E.g. if we upgrade (in the release after) from H2 version 2.x to 2.x we must not set the "FROM_1X flag".